pkg/network/ovn_kubernetes.go

```diff
@@ -174,7 +174,24 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 	}
 	objs = append(objs, manifests...)
 
-	updateNode, updateMaster := shouldUpdateOVNK(bootstrapResult.OVN.ExistingNodeDaemonset, bootstrapResult.OVN.ExistingMasterDaemonset, os.Getenv("RELEASE_VERSION"))
+	// obtain the current IP family mode.
+	ipFamilyMode := names.IPFamilySingleStack
+	if len(conf.ServiceNetwork) == 2 {
+		ipFamilyMode = names.IPFamilyDualStack
+	}
+	// check if the IP family mode has changed and control the conversion process.
+	updateNode, updateMaster := shouldUpdateOVNKonIPFamilyChange(bootstrapResult.OVN.ExistingNodeDaemonset, bootstrapResult.OVN.ExistingMasterDaemonset, ipFamilyMode)
+	// annotate the daemonset and the daemonset template with the current IP family mode,
+	// this triggers a daemonset restart if there are changes.
+	err = setOVNDaemonsetAnnotation(objs, names.NetworkIPFamilyModeAnnotation, ipFamilyMode)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to set IP family %s annotation on daemonsets", ipFamilyMode)
+	}
+
+	// don't process upgrades if we are handling a dual-stack conversion.
+	if updateMaster && updateNode {
```

Comment:
@aojea do you mean `!updateMaster && !updateNode` here instead? 

If I read the code right, then at this point one or both of updateNode and updateMaster will be true when we need to dual-stack update. But in that case, we *don't* want to do the master/node version upgrades right?

Reply:
@aojea also any chance you can do a testcase that test both the version upgrade and the dual-stack conversion at the same time, so that we can ensure they work correctly together?

Reply:
yeah, sure

Reply:
It took me a while to understand at first too, updateMaster to true means ok, is an inverse logic, you hold the updates , so it means that you are doing something only when one is false

Reply:
Ok yeah, I see that now; I needed to look at more than just the diff.

Reply:
@aojea do you want to add the test that @dcbw mentioned for conversion and upgrade at same time in this PR or do it later in a separate PR?

Reply:
It should merge together, will add it later today or tomorrow